src/App.tsx

```diff
@@ -1,61 +1,118 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const getNewId = (list: Todo[]) => {
+  return Math.max(...list.map((todo) => todo.id)) + 1;
+};
+
+export const App: React.FC = () => {
+  const [list, setList] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState('');
+  const [titleIsCorrect, setTitleIsCorrect] = useState(true);
+  const [userIsCorrect, setUserIsCorrect] = useState(true);
+
+  const submitHandler = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!title.trim()) {
+      setTitleIsCorrect(false);
+    }
+
+    if (!user) {
+      setUserIsCorrect(false);
+    }
+
+    if (title.trim() && user) {
+      setList(
+        [...list, {
+          id: getNewId(list),
+          userId: +user,
+          title,
+          completed: false,
+          user: getUser(+user),
+        }],
+      );
+
+      setTitle('');
```

Comment:
create reset function for this

Reply:
agree, will be better to create separate function for reset form functionality